cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/impl/CassandraSequenceHandler.java

```diff
@@ -0,0 +1,155 @@
+		primaryKeyName.add( "sequence_name" );
+		List<String> columnNames = new ArrayList<String>( 2 );
+		columnNames.add( "sequence_name" );
+		columnNames.add( "sequence_value" );
+		List<String> columnTypes = new ArrayList<String>( 2 );
+		columnTypes.add( "varchar" );
+		columnTypes.add( "bigint" );
+		datastoreProvider.createColumnFamilyIfNeeded( generatorsKey, primaryKeyName, columnNames, columnTypes );
+	}
+
+	private Long nextValueSelect(String tableName, String sequenceName) {
+
+		Statement select = select().column( "sequence_value" ).from( "\"" + tableName + "\"" )
+				.where( eq( "sequence_name", QueryBuilder.bindMarker() ) );
+
+		PreparedStatement preparedStatement = provider.getSession().prepare( select.toString() );
+		BoundStatement boundStatement = preparedStatement.bind( sequenceName );
+
+		ResultSet resultSet;
+		try {
+			resultSet = provider.getSession().execute( boundStatement );
+		}
+		catch (DriverException e) {
+			System.out.println( e.toString() );
+			throw e;
+		}
+
+		if ( resultSet.isExhausted() ) {
+			return null;
+		}
+		else {
+			return resultSet.one().getLong( 0 );
+		}
+	}
+
+	private Long nextValueInsert(String tableName, String sequenceName, Long value) {
+
+		Insert insert = insertInto( "\"" + tableName + "\"" )
+				.value( "sequence_name", QueryBuilder.bindMarker( "sequence_name" ) )
+				.value( "sequence_value", QueryBuilder.bindMarker( "sequence_value" ) )
+				.ifNotExists();
+
+		PreparedStatement preparedStatement = provider.getSession().prepare( insert.toString() );
+		BoundStatement boundStatement = preparedStatement.bind();
+		boundStatement.setString( "sequence_name", sequenceName );
+		boundStatement.setLong( "sequence_value", value );
+
+		try {
+			provider.getSession().execute( boundStatement );
+		}
+		catch (DriverException e) {
+			System.out.println( e.toString() );
+			throw e;
+		}
+
+		return nextValueSelect( tableName, sequenceName );
+	}
+
+	private boolean nextValueUpdate(String tableName, String sequenceName, Long oldValue, Long newValue) {
+
+		Statement update = update( "\"" + tableName + "\"" )
+				.with( set( "sequence_value", QueryBuilder.bindMarker( "sequence_value_new" ) ) )
+				.where( eq( "sequence_name", QueryBuilder.bindMarker( "sequence_name" ) ) )
+				.onlyIf( eq( "sequence_value", QueryBuilder.bindMarker( "sequence_value_old" ) ) );
+
+		PreparedStatement preparedStatement = provider.getSession().prepare( update.toString() );
+		BoundStatement boundStatement = preparedStatement.bind();
+		boundStatement.setString( "sequence_name", sequenceName );
+		boundStatement.setLong( "sequence_value_new", newValue );
+		boundStatement.setLong( "sequence_value_old", oldValue );
+
+		ResultSet resultSet;
+		try {
+			resultSet = provider.getSession().execute( boundStatement );
+		}
+		catch (DriverException e) {
+			System.out.println( e.toString() );
+			throw e;
+		}
+
+		return resultSet.one().getBool( 0 );
+	}
+
+	public Number nextValue(NextValueRequest request) {
```

Comment:
Did you look into [Cassandra counters](http://www.datastax.com/dev/blog/whats-new-in-cassandra-2-1-a-better-implementation-of-counters) by any chance? Seems like we could use those to represent sequences?